libglusterfs/src/event-epoll.c

```diff
@@ -114,9 +115,10 @@ __event_slot_alloc(struct event_pool *event_pool, int fd,
 
     for (j = 0; j < EVENT_EPOLL_SLOTS; j++) {
         if (table[j].fd == -1) {
-            /* wipe everything except bump the generation */
+            /* Bump the generation and wipe everything except lock and
+               list because they need an explicit initialization anyway. */
             gen = table[j].gen;
-            memset(&table[j], 0, sizeof(table[j]));
+            memset(&table[j], 0, offsetof(struct event_slot_epoll, lock));
```

Comment:
This is so much dependent on structure definition and bit dangerous.
What if somebody adds some fields at the end of the structure in future?

Reply:
I agree with Ashish. We should avoid that, even if use a compile time check. I would simply do nothing here. I would move the responsibility to the caller of this function, who should correctly initialize all fields.